internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -608,6 +643,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		if kerrors.IsConflict(err) {
 			return reconcile.Result{Requeue: true}, nil
 		}
+
+		meta := handleCommonCompositionResult(ctx, r, res, xr, log)
+		for _, c := range xr.GetConditions() {
+			if xpv1.IsSystemConditionType(c.Type) {
+				continue
+			}
+			if _, ok := meta.conditionTypesSeen[c.Type]; !ok {
+				c.Status = corev1.ConditionUnknown
+				c.Reason = reasonPriorFailure
+				c.Message = ""
+				xr.SetConditions(c)
+			}
+		}
```

Comment:
I'd add a comment explaining what's happening here. It took me a moment to connect the dots.

Also, I found `PriorFailure` as a reason a little confusing. I think you mean "a failure happened prior to determining this condition's status", but it read to me at first as "this condition had a prior failure".

That said, I'm struggling to come up with a better name. 🤔 Something that says "Crossplane couldn't reach the code that updates this condition because it hit a fatal error." Maybe just `FatalError`? I guess we could include what I just wrote in the message to make it extra clear.

Reply:
Yeah I was thinking of commenting this bit, I should have done it 😄.
I changed this to `FatalError` and added the message as well. LMK what you think.